src/cdk-experimental/menu/context-menu.ts

```diff
@@ -0,0 +1,318 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  Directive,
+  Input,
+  ViewContainerRef,
+  Output,
+  EventEmitter,
+  Optional,
+  OnDestroy,
+  Inject,
+  Injectable,
+  InjectionToken,
+} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {Directionality} from '@angular/cdk/bidi';
+import {
+  OverlayRef,
+  Overlay,
+  OverlayConfig,
+  FlexibleConnectedPositionStrategy,
+  ConnectedPosition,
+} from '@angular/cdk/overlay';
+import {TemplatePortal} from '@angular/cdk/portal';
+import {coerceBooleanProperty, BooleanInput} from '@angular/cdk/coercion';
+import {fromEvent, merge, Subject} from 'rxjs';
+import {take, takeUntil} from 'rxjs/operators';
+import {CdkMenuPanel} from './menu-panel';
+import {MenuStack, MenuStackItem} from './menu-stack';
+
+/**
+ * Check if the given element is part of the cdk menu module or nested within a cdk menu element.
+ * @param target the element to check.
+ * @return true if the given element is part of the menu module or nested within a cdk menu element.
+ */
+function isWithinMenuElement(target: Element | null) {
+  while (target instanceof Element) {
+    if (target.className.indexOf('cdk-menu') !== -1) {
```

Comment:
This can break on something like `class="cdk-menu-close"`. You should use `target.classList.contains('cdk-menu')` instead.

Reply:
All cdk menu classes are in the form of `cdk-menu-*` so we want to check if we've hit any directive not just the `cdk-menu` directive. However I could see users placing `cdk-menu-foo` on their own elements but not sure how likely that is. @jelbourn thoughts?

Reply:
My stance is that we own the `cdk-` prefix and users shouldn't be making up their own cdk classes. If this breaks as a result of that, it's outside the scope of our purview.